### DISCUSSION TITLE
JCB layout maybe bug

### DISCUSSION THREAD

**Discussion by saman222** [joomengine/Joomla-Component-Builder#647] [Q&A] [answered]
Hi
when i create some layouts (default , ...) then we want use that in our site view of component, we must copy it's syntax from "Layout Code Snippets" (e.x <?php echo JLayoutHelper::render('default', $this->item); ?> and other snippets ) to "Default Tmpl" in the site view to put all layout files (default.php , .... ) to .../components/com_mycomponent/layouts/ .

but the problem: if i have multi layout i need to copy all syntax from Layout Code Snippets to "Default Tmpl" textbox and if do this in the site view we see all layouts in one siteview page. 

out temporary solution: we handle this problem by use // in the <?php // echo JLayoutHelper::render('default', $this->item); ?> to put file in the layouts folder but not render in the siteview. we use multi layout because we want user can select each layout to render the siteview.

**Comment by Llewellynvdm** [comment, marked as answer]
I am not sure if I understand your problem, but just to be sure we are on the same page, have you watched (forgive me if you did) but have you watched the tutorials about using layouts, and templates in site views?

Since being able to add a layout to a site view is very easy, you just add the code link this.

For a template:
```php
<?php echo $this->loadTemplate('programs'); ?>
```

For a layout:
```php
<?php echo JLayoutHelper::render('signup', $values_passed_to_layout); ?>
```

Then JCB will automatically add those templates or layouts to your component. This also means that if you would like a layout in your component, you can with the following convention add it without it rendering, but it still gets added:
```php
<?php  // JLayoutHelper::render('signup', $values_passed_to_layout); ?>
```

Please let me know what you are trying something beyond this basic implementation, if this does not resolve or answer your question.

**Reply by saman222** [reply]
hi and thank you @Llewellynvdm .
i think if use // to comment line to prevent render layout is a bug , not my solution but it seems it is a solution.
anyway i have an other question that i don't understand can ask here or need to create new discussion : if i want put my layouts to different folders, for example i have some layouts for TeacherSiteView and some layouts for CourseSiteView and get capability to joomlasuperuser to select his desired layout for specific site view. 
how i can put different layouts in different path? for example put specific layouts in each these path:
.../components/com_mycomponent/layouts/teacherlayout 
and
.../components/com_mycomponent/layouts/courselayout

**Reply by Llewellynvdm** [reply]
I would add a suffix, or prefix to the layout name to achieve that kind of list selection option. But by default JCB does not add the layouts into more folders for site views, only for admin views at this time.

You should look at how we managed layout selections in sermon distributor to see the convenient path of multi layout options, and selecting those with global settings.
